src/packages/engine-core/src/NAPIEngine.ts

```diff
@@ -428,56 +437,53 @@ You may have to run ${chalk.greenBright(
     transaction = false,
     numTry = 1,
   ): Promise<any> {
-    await this.start()
     debug('requestBatch')
     const headers: QueryEngineRequestHeaders = {}
     const request: QueryEngineBatchRequest = {
       batch: queries.map((query) => ({ query, variables: {} })),
       transaction,
     }
+    await this.start()
```

Comment:
why did we move this down here if it does not require any of the variables?

Reply:
Because it seems to fix the flaky `Engine Not Yet Connect` error